src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,8 +34,24 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        
+        String[] addressParts = trimmedAddress.split(",");
+        this.block = new Block(addressParts[0].trim().substring(2));
+        this.street = new Street(addressParts[1].trim());
+        if (addressParts.length > 2) {
+            this.unit = new Unit(addressParts[2].trim());
+            if (addressParts.length > 3)
+                this.postCode = new PostalCode(addressParts[3].trim());
+            else this.postCode = new PostalCode("");
+        }
+        
+        else {
```

Comment:
coding standard violation!
`else` should be cuddled between the braces as per our coding guidelines